app_server.py

```diff
@@ -182,10 +183,14 @@ def handle(self, exception):
                 sys.exit()
         return make_response(body, 500, {"Content-Type": "text/plain"})
 
+
 class HeartbeatController(object):
 
     def heartbeat(self):
-        return make_response("", 200, {"Content-Type": "text/plain"})
+        server_info = json.dumps(dict(
+            git_ref=subprocess.check_output(['git', 'rev-parse', 'HEAD']).strip()
```

Comment:
I'm a little concerned that someone who knows the heartbeat URL can spawn a bunch of processes on the server. What if this information were obtained once and cached as a class variable, or passed in to the HeartbeatController constructor?